File: yujin-level0/Week62/세 개의 구분자.js
function solution(myStr) {
  // 모든 구분자를 'a' 하나로 합침
  let UnifiedStr = myStr.replaceAll("b", "a").replaceAll("c", "a");
  let answer = UnifiedStr.split("a").filter((str) => str !== "");

  return answer.length === 0 ? ["EMPTY"] : answer;
}

// 다른 풀이
// const solution=s=>s.match(/[^a-c]+/g)||['EMPTY']

File: gnsals0904/level0/세_개의_구분자.java
import java.util.*;

class Solution {
    public String[] solution(String myStr) {
        char[] myStrToChar = myStr.toCharArray();
        StringBuilder sb = new StringBuilder();
        ArrayList<String> answer = new ArrayList<>(1_000_000);
        for(char myChar : myStrToChar){
            if(isABC(myChar)){
                if(!sb.isEmpty()){
                    answer.add(sb.toString());
                    sb.setLength(0);
                }
                continue;
            }
            sb.append(myChar);
        }

        if (!sb.isEmpty()) {
            answer.add(sb.toString());
        }

        return answer.isEmpty()
                ? new String[]{"EMPTY"}
                : answer.toArray(String[]::new);
    }

    static public boolean isABC(char c){
        return (c == 'a' || c == 'b' || c == 'c');
    }
}